File: 2358-number-of-ways-to-split-array/2358-number-of-ways-to-split-array.go
func waysToSplitArray(nums []int) int {
	ans := 0
	left_sum := 0
	total := 0

	for i := 0; i < len(nums); i++ {
		total += nums[i]
	}
	for i := 0; i < len(nums)-1; i++ {
		left_sum += nums[i]
		right_sum := total - left_sum

		if left_sum >= right_sum {
			ans++
		}
	}

	return ans
}

File: 2358-number-of-ways-to-split-array/2358-number-of-ways-to-split-array.py
class Solution:
    def waysToSplitArray(self, nums: List[int]) -> int:
        length = len(nums)
        leftSum = [0] * (length - 1)
        rightSum = [0] * (length - 1)

        for i in range(length - 1):
            if i == 0:
                leftSum[i] = nums[i]
            else:
                leftSum[i] = leftSum[i - 1] + nums[i]

        count = 0
        for i in range(-1, -length, -1):
            if i == -1:
                rightSum[i] = nums[i]
            else:
                rightSum[i] = nums[i] + rightSum[i + 1]

            if leftSum[i] >= rightSum[i]: count += 1
        
        return count